fitsio/hdu/base.py

```diff
@@ -320,6 +321,12 @@ def write_keys(self, records_in, clean=True):
             if name is not None:
                 name = name.upper()
 
+                if any(c in INVALID_HDR_CHARS for c in name):
```

Comment:
this is potentially slow in python when writing big headers.  We should test to see if this needs optimization, e.g. compiled regexp or check in C